src/main/scala/io/delta/DeltaMergeBuilder.scala

```diff
@@ -132,31 +379,107 @@ case class DeltaMergeMatchedActionBuilder private[delta](
+      matchCondition: Option[Column]): DeltaMergeMatchedActionBuilder = {
+    new DeltaMergeMatchedActionBuilder(mergeBuilder, matchCondition)
+  }
+}
+
+
+/**
+ * :: Evolving ::
+ *
+ * Builder class to specify the actions to perform when a source row has not matched any target
+ * Delta table row based on the merge condition, but has matched any additional condition
+ * if specified.
+ *
+ * See [[DeltaMergeBuilder]] for more information.
+ *
+ * @since 0.3.0
+ */
+@Evolving
+class DeltaMergeNotMatchedActionBuilder private(
     private val mergeBuilder: DeltaMergeBuilder,
-    private val matchCondition: Option[Column]) {
+    private val notMatchCondition: Option[Column]) {
 
+  /**
+   * :: Evolving ::
+   *
+   * Insert a new target Delta table row based on the rules defined by `values`.
+   *
+   * @param values rules to insert a row as a Scala map between target column names and
+   *               corresponding expressions as Column objects.
+   * @since 0.3.0
+   */
+  @Evolving
   def insert(values: Map[String, Column]): DeltaMergeBuilder = {
     addInsertClause(values)
   }
 
+  /**
+   * :: Evolving ::
+   *
+   * Insert a new target Delta table row based on the rules defined by `values`.
+   *
+   * @param values rules to insert a row as a Scala map between target column names and
+   *               corresponding expressions as SQL formatted strings.
+   * @since 0.3.0
+   */
+  @Evolving
   def insertExpr(values: Map[String, String]): DeltaMergeBuilder = {
     addInsertClause(toStrColumnMap(values))
   }
 
+  /**
+   * :: Evolving ::
+   *
+   * Insert a new target Delta table row based on the rules defined by `values`.
+   *
+   * @param values rules to insert a row as a Java map between target column names and
+   *               corresponding expressions as Column objects.
+   * @since 0.3.0
+   */
+  @Evolving
   def insert(values: java.util.Map[String, Column]): DeltaMergeBuilder = {
     addInsertClause(values.asScala)
   }
 
+  /**
+   * :: Evolving ::
+   *
+   * Insert a new target Delta table row based on the rules defined by `values`.
+   *
+   * @param values rules to insert a row as a Java map between target column names and
+   *               corresponding expressions as SQL formatted strings.
+   *
+   * @since 0.3.0
+   */
+  @Evolving
   def insertExpr(values: java.util.Map[String, String]): DeltaMergeBuilder = {
     addInsertClause(toStrColumnMap(values.asScala))
   }
 
+  /**
+   * :: Evolving ::
+   *
+   * Insert a new target Delta table row by assigning the target columns to the values of the
+   * corresponding columns in the source row.
+   * @since 0.3.0
+   */
+  @Evolving
   def insertAll(): DeltaMergeBuilder = {
     val insertClause = MergeIntoInsertClause(
-      matchCondition.map(_.expr),
+      notMatchCondition.map(_.expr),
       MergeIntoClause.toActions(Nil, Nil))
-    mergeBuilder.copy(whenClauses = mergeBuilder.whenClauses :+ insertClause)
+    mergeBuilder.withClause(insertClause)
```

Comment:
what is this change about?

Reply:
because i changed from case class to simple class, there were no more copy() methods.